client/public/embed.js

```diff
@@ -0,0 +1,96 @@
+"use strict";
+
+var onReady = function(cb) {
+  // https://plainjs.com/javascript/events/running-code-when-the-document-is-ready-15/
+  // in case the document is already rendered
+  if (document.readyState != "loading") cb();
+  // modern browsers
+  else if (document.addEventListener)
+    document.addEventListener("DOMContentLoaded", cb);
+  // IE <= 8
+  else
+    document.attachEvent("onreadystatechange", function() {
+      if (document.readyState == "complete") cb();
+    });
+};
+
+var url =
+  document.location.hostname == "localhost"
+    ? "http://localhost:3000"
+    : "https://sketch.sh";
+
+var Sketch = {
+  defaultHeight: "200px",
+  defaultLang: "RE",
+  defaultAutoHeight: true,
+  onLoad: function() {
+    var elements = document.querySelectorAll("[data-sketch]");
```

Comment:
Small tweak — could we have `createEmbed` work for collections as well?

My thinking is that I'd like to try this out with _all code snippets in a page_ and _the current `odoc` output_, which doesn't include these `[data-sketch]` tags.

So essentially, I'd find it more useful to have:

```html
<body>
  <!-- generated docs goes all in here -->
  <script src="https://sketch.sh/embed.js"></script>
    <script>
      Sketch.createEmbeds({
        elementQuery: () =>
          [...document.getElementByTagName("code")]
             .filter( x => x.className.match(/^re hljs ocaml$/) ),
        height: "300px",
      });
    </script>
</body>
```

Since it can automatically use the body (text) value of whatever elements matched that query too.

100% _Drop-in_.

Reply:
Alternatively passing in a function to extract the source of the code would work great too:

```js
Sketch.createEmbeds({
        elementQuery: () =>
          [...document.getElementByTagName("code")]
             .filter( x => x.className.match(/^re hljs ocaml$/) ),
        source: $el => $el.innerText,
        height: "300px",
      });
```

Reply:
wouldn't this be a matter of calling createEmbed on the colletion yourself? This would reduce complexity that I need to maintain. Also, we need access to original code, not the highlighted ones